app/src/main/java/com/vero/tasky/agenda/data/broadcastreceiver/BootCompletedReceiver.kt

```diff
@@ -15,7 +16,7 @@ class BootCompletedReceiver : BroadcastReceiver() {
 
     override fun onReceive(context: Context?, intent: Intent?) {
         if (intent?.action == Intent.ACTION_BOOT_COMPLETED) {
-            alarmHandler.updateAlarmsForAllAgendaItems()
+            runBlocking { alarmHandler.setAlarmForFutureAgendaItems() }
```

Comment:
Another way, is to run a CoroutineScope here
So you run it on a coroutine, and not on the main thread

This is an extension function I use for this cases

```
fun BroadcastReceiver.goAsync(
    context: CoroutineContext = EmptyCoroutineContext,
    block: suspend CoroutineScope.() -> Unit
) {
    val pendingResult = goAsync()
    CoroutineScope(SupervisorJob()).launch(context) {
        try {
            block()
        } finally {
            pendingResult.finish()
        }
    }
}
```

And then I use it like:

```
override fun onReceive(context: Context?, intent: Intent?) = goAsync {
        if (context == null || intent == null) return@goAsync
        if (intent.action != Intent.ACTION_BOOT_COMPLETED) return@goAsync
        alarmHandler.setAlarmForFutureAgendaItems()
    }
```